Some mimic changes

loqoman opened the pull request:
([I WAS HIDING](https://www.youtube.com/watch?v=Fm8PkRWTnns))

:cl: loqoman
tweak: Mimics have been tweaked a bit to be more like...Mimics!
/:cl:

More of a learning project if anything, this is going to be a *slight* rework of mimics!

What has been done:

- Mimic's now have an ability that changes their sprite, and has them lay dormant (But they can still move) 
- Mimic's have some slightly better sprites for when they are aggressive, as well as sprites for the ability.

What has yet to be done:

- Finishing the mimic ability. The final idea is for the ability is to have the mimic disguise as a crate, and when crewmembers interact with the "Crate" the mimic will go full rage mode, and deal a large amount of damage to the person that touched it.
- Various cleanup(Making the visible_message nicer, cleaning up sprite timings, ect.)

As you can see, this is not a completed project, but considering this is my first contribution, I want to get it out there for QA/QC, as well as additional design ideas if people have them. 
![image](https://user-images.githubusercontent.com/14814137/30359270-72a9ee8e-97fe-11e7-9541-dbc47f141ca6.png) 
(Sprite)

Commit:
Some working mimic changes

I WAS HIDING

RemieRichards commented on code/modules/mob/living/simple_animal/hostile/mimic.dm:
```diff
@@ -73,8 +77,8 @@
 /mob/living/simple_animal/hostile/mimic/crate/proc/trigger()
-	if(!attempt_open)
-		visible_message("<b>[src]</b> starts to move!")
+	if(!attempt_open) //Have not attempted to open
```
comments like this are worthless, remove all comments you've added.

loqoman replied:
Alright, what kind of comments SHOULD I be writing?

RemieRichards commented on code/modules/mob/living/simple_animal/hostile/mimic.dm:
```diff
@@ -266,3 +270,22 @@ GLOBAL_LIST_INIT(protected_objects, list(/obj/structure/table, /obj/structure/ca
+/datum/action/innate/mimic/root/Activate()
+	var/mob/living/simple_animal/hostile/mimic/crate/S = owner
+
+	if(S.attempt_open == TRUE) //If we have tried to open it
```
just ```if(S.attempt_open)``` is preferred.

RemieRichards commented on code/modules/mob/living/simple_animal/hostile/mimic.dm:
```diff
@@ -266,3 +270,22 @@ GLOBAL_LIST_INIT(protected_objects, list(/obj/structure/table, /obj/structure/ca
+
+
+	//TBD, but see trellio
+	return
```
a return at the end of a function that does nothing is a trailing return.
It's unneeded, and should be removed.

RemieRichards commented on code/modules/mob/living/simple_animal/hostile/mimic.dm:
```diff
@@ -266,3 +270,22 @@ GLOBAL_LIST_INIT(protected_objects, list(/obj/structure/table, /obj/structure/ca
 	icon_state = TrueGun.icon_state
 	icon_living = TrueGun.icon_state
+
+/datum/action/innate/mimic/root
```
this shouldn't be ```/innate```

loqoman replied:
Okay, what types of actions are supposed to be innate?

RemieRichards commented on code/modules/mob/living/simple_animal/hostile/mimic.dm:
```diff
@@ -266,3 +270,22 @@ GLOBAL_LIST_INIT(protected_objects, list(/obj/structure/table, /obj/structure/ca
+	button_icon_state = "mimicA"
+	desc = "Roots yourself in place and disguises you as a normal crate. Anyone to try to open you up is given a nasty surprise!"
+
+/datum/action/innate/mimic/root/Activate()
```
this should be ```Trigger()```, with the first line of it being:
```
if(!..())
    return 0
```

InsaneHyena commented:
We have mimics?

Mey-Ha-Zah commented:
Thought they were called Morphs.

RemieRichards commented:
Morphs != Mimics.
Two completely different sections of code.

Supermichael777 commented:
mimics are just a tragically underused fake crate.

Bawhoppen commented:
I wanted to add loot drops to them though and now you walk in and do this

loqoman commented:
@RemieRichards It's a very basic simple mob that acts as a fake crate, and triggers when people attempt to open it.

loqoman commented:
morps are the mob that eats things to disguise as them

RemieRichards commented:
I Know?? I'm telling THEM that.

loqoman commented:
Eh? Oh sorry, read it wrong. MB

loqoman commented:
@Bawhoppen What loot drops were you thinking about?

Bawhoppen commented:
Well nothing super specific, but I just wanted them to drop something lootwise to fit with the theme of how mimics sometimes have loot inside them after you slay them in other lore,

loqoman committed:
Remie's requests.

Removes trailing return, removes my comments, and removes (S.attempt_open == TRUE)

loqoman committed:
Merge branch 'master' into Mimic_Rework

Armhulen commented:
i agree with you on the loot thing but it's a xenobio mob so you can't really do that.

Armhulen commented:
also you can't make a mob work like an object so why not have root make an object that when interacted with opens and does the whole mimic deal

WJohn commented:
This sprite is terrible but at least there is now a clear indication it is a minimal. Sprite can be improved later.

loqoman commented:
I'm a lazy fuck!
